base/applications/rapps/include/rosui.h

```diff
@@ -302,54 +300,78 @@ class CUiBox
         *newRect = currentRect;
     }
 
-
 public:
+    //virtual ~CUiBox() {} // Needed for current CUiCollection implementation.
+
+    // Override in subclass
     virtual VOID ComputeMinimalSize(SIZE* size)
     {
-        // Override in subclass
         size->cx = max(size->cx, 0);
         size->cy = min(size->cy, 0);
-    };
+    }
 
+    // Override in subclass
     virtual VOID ComputeContentBounds(RECT* rect)
     {
-        // Override in subclass
-    };
+    }
 
-    virtual DWORD_PTR CountSizableChildren()
+    // Override in subclass
+    virtual INT CountSizableChildren()
     {
-        // Override in subclass
         return 0;
-    };
+    }
 
+    // Override in subclass
     virtual HDWP OnParentSize(RECT parentRect, HDWP hDwp)
     {
-        // Override in subclass
-        return NULL;
-    };
+        /* Pass hDwp down the call chain */
+        return hDwp;
+    }
 };
 
-class CUiPrimitive
+
+
+#if 1
+//
+// This is actually an interface of sorts...
+// Actually it could be helpful if it's a reference count
+// so that the UiCollection could implement the ref uncount and delete if needed.
+//
+class CUiPrimitive : public CUiBox
 {
 protected:
-    CUiPrimitive * m_Parent;
+#if 0
+    CUiPrimitive* m_Parent;
+#endif
 
 public:
     virtual ~CUiPrimitive() {}
 
-    virtual CUiBox * AsBox() { return NULL; }
+    // virtual CUiBox* AsBox() { return NULL; }
 };
+#else
+#define CUiPrimitive CUiBox
+#endif
 
 class CUiCollection :
-    public CPointerArray < CUiPrimitive >
+    public CPointerArray</*CUiCollection,*/ CUiPrimitive>
 {
-    virtual BOOL OnRemoveItem(CUiPrimitive * ptr)
+public:
+    virtual ~CUiCollection() {}
+
+#if 0
+    /*virtual*/ BOOL OnRemoveItem(CUiPrimitive* ptr)
     {
         delete ptr;
         return TRUE;
     }
+#endif
 };
 
+//
+// Question: Does it need more? If not, we could just use
+// CUiCollection's directly, in lieu of CUiContainer's.
+//
```

Comment:
So yeah, why do we need to have this distinction of UI "container" and UI "collection" ?
Here it seems apparent they would be the same. What was the reason behind this distinction @gigaherz ?